common/src/test/java/com/thoughtworks/go/domain/MaterialRevisionsTest.java

```diff
@@ -487,6 +488,33 @@ public void shouldUseUpstreamPipelineLabelForDependencyMaterial() {
         assertThat(namedRevisions.get(pipelineName), is(pipelineLabel));
     }
 
+    @Test
+    public void shouldPopulateEnvironmentVariablesForEachOfTheMaterialsWithTheValueForChangedFlag(){
+        EnvironmentVariableContext context = new EnvironmentVariableContext();
+        MaterialRevisions revisions = new MaterialRevisions();
+        revisions.addRevision(new MaterialRevision(MaterialsMother.hgMaterial("empty-dest-and-no-name", null), true, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.hgMaterial("url", "hg-folder1"), true, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.hgMaterial("url", "hg-folder2"), false, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.dependencyMaterial("p1", "s1"), true, ModificationsMother.changedDependencyMaterialRevision("p3", 1, "1", "s", 1, new Date()).getModifications()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.dependencyMaterial("p2", "s2"), false, ModificationsMother.changedDependencyMaterialRevision("p3", 1, "1", "s", 1, new Date()).getModifications()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.pluggableSCMMaterial("scm1", "scm1name"), true, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.pluggableSCMMaterial("scm2", "scm2name"), false, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.packageMaterial("repo1", "repo1name", "pkg1", "pkg1name"), true, ModificationsMother.multipleModificationList()));
+        revisions.addRevision(new MaterialRevision(MaterialsMother.packageMaterial("repo2", "repo2name", "pkg2", "pkg2name"), false, ModificationsMother.multipleModificationList()));
+
+        revisions.populateEnvironmentVariables(context, null);
+
+        assertThat(context.getProperty("GO_MATERIAL__HAS_CHANGED"), is("true"));
```

Comment:
FYI: Without the change [here](https://github.com/gocd/gocd/pull/4220/files#diff-cdff4287c76672b79e5c4c818b84b689R156), that variable becomes: `GO_MATERIAL_NULL_HAS_CHANGED`.